tiledb/sm/rest/curl.cc

```diff
@@ -241,6 +258,27 @@ Status Curl::init(
   return Status::Ok();
 }
 
+size_t Curl::OnReceiveData(
+    void* res_data, size_t tSize, size_t tCount, void* userdata) {
+  size_t length = tSize * tCount, index = 0;
+  while (index < length) {
+    unsigned char* temp = (unsigned char*)res_data + index;
+    if ((temp[0] == '\r') || (temp[0] == '\n'))
+      break;
+    index++;
+  }
+
+  std::string str((unsigned char*)res_data, (unsigned char*)res_data + index);
```

Comment:
which `std::string` constructor is this expected to call? I don't understand the second argument, it appears to be pointing to either a `'\r'` or `'\n'` character. Was this suppose to be the size, maybe just `index`?

Reply:
It uses the constructor of std::string that takes this strings as `InputIterators` _beg & _end accordingly. The index is calculated without `/r` and `/n` by the loop above. No worries I will change it to be clearer.

Reply:
An input iterator would be of type `std::string::iterator`, these inputs are `unsigned char*`.